### DISCUSSION TITLE
API source maps

### DISCUSSION THREAD

**Discussion by davidkaneda** [vercel/next.js#23469] [Help]
I'm currently setting up the Sentry integration, hosted on Vercel, based on the [with-sentry](https://github.com/vercel/next.js/tree/canary/examples/with-sentry) example. Client-side source maps work great with the `productionBrowserSourceMaps` option in `next.config.js`, but I currently get the compiled file when using `Sentry.captureException` in an API route — It would be great to be able to see accurate line numbers for exceptions in the API, and further, to link it to the codebase using Sentry's repository mapping feature.

Has anyone gotten this to work?

For reference, here's what a client-side error looks like:

![image](https://user-images.githubusercontent.com/21713/112740461-1053db80-8f32-11eb-846a-2798249fdc88.png)

vs. an API error:

![image](https://user-images.githubusercontent.com/21713/112740475-324d5e00-8f32-11eb-9fde-cfb44c201713.png)

Also:
- I saw there's a [next-source-maps](https://github.com/vercel/next-plugins/tree/master/packages/next-source-maps) plugin, which I read somewhere handles server-side source maps, but it's 2 years old and I believe I saw somewhere that it's deprecated
- I tried just setting `config.devtool = "source-map"`  in `next.config.js`, which does actually generate source maps where there weren't any before in `.next/server/pages/api` during `next build`, but it doesn't appear this actually sent them to Sentry when run on Vercel

Many thanks!

**Comment by CaptainChemist** [comment]
Thanks for posting this, I thought I was losing my mind because there were several examples of Nextjs with Sentry but I could never figure out how to get the API routes to actually show the real code in the error. Did you figure anything out?

**Reply by davidkaneda** [reply]
Unfortunately not :/

**Comment by t4t5** [comment]
Has anyone figured out a workaround for this yet? 🤔

**Reply by Merott** [reply]
Almost a year later, I'm struggling with the same as I'm trying to configure Sentry manually without the webpack plugin.

Any updates in the past year? 🤔
